spec/ruby/core/string/gsub_spec.rb

```diff
@@ -225,6 +227,64 @@ def replacement.to_str() "hello_replacement" end
     result.should == 'éué'
     result.encoding.should == Encoding::UTF_8
   end
+
+  describe "when both pattern and replacement are the same byte-length" do
```

Comment:
What differs than the different byte-length cases?
Please add regression tests to `test/ruby/`.